File: Proiect/src/main/java/repository/Repository.java
package repository;

import java.util.Collection;

public interface Repository<T, Tid> {
    void add(T el);
    void delete(T el);
    void update(T el,Tid id);
    T findById(Tid id);
    Iterable<T> findAll();
    Collection<T> getAll();
}

File: ProiectCsharp/MPP/MPP/repository/Repository.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace MPP.repository
{
    internal interface Repository<T, Tid>
    {
        void add(T entity);
        void delete(T entity);
        void update(T entity,Tid id);
        T findbyId(Tid id);

        ICollection<T> findAll();
    }
}
